tests/RobotFramework/tests/cumulocity/shell/shell_operation.robot

```diff
@@ -29,6 +29,11 @@ Failed shell command
     ${operation}=    Cumulocity.Execute Shell Command    exit 1
     Operation Should Be FAILED    ${operation}
 
+Shell command succeeds if output is too large
+    [Documentation]    Output should be trimmed by c8y mapper.
+    ${operation}=    Cumulocity.Execute Shell Command    yes 'hello"' | head -n 100000
+    Operation Should Be SUCCESSFUL    ${operation}
+
 
```

Comment:
After the operation succeeds, I would add a validation step to check the size of the output

Reply:
I think the actual size is an implementation detail. From the user's perspective, it doesn't matter what the size of the message is, if they e.g. launched some command that generated massive output, but they care that the operation is marked SUCCESSFUL, even if the output is massive.
But as said in https://github.com/thin-edge/thin-edge.io/pull/3179#pullrequestreview-2373065149, a user might want to know if the message was truncated, so I'm happy to add an assert for that case.

Reply:
Something like 
```
${output_length}=    Get Length of Output    ${operation}
Should Be Less Than    ${output_length}    50000    # Assuming trimming occurs above 50,000 characters.
```

Reply:
> Something like
> 
> ```
> ${output_length}=    Get Length of Output    ${operation}
> Should Be Less Than    ${output_length}    50000    # Assuming trimming occurs above 50,000 characters.
> ```

This won't be possible as this is checking the operation in the json format will will be different to the length of the SmartREST 2.0 MQTT message. So doing this check won't yield any useful results.

The fact that operation is set to SUCCESSFUL is proof that the message is under Cumulocity's max MQTT payload size.